core/src/main/java/org/mule/runtime/core/internal/processor/strategy/reactor/builder/ReactorPublisherBuilder.java

```diff
@@ -201,6 +220,18 @@ private FluxBuilder(Flux<CoreEvent> flux) {
       return this;
     }
 
+    @Override
+    public ReactorPublisherBuilder<Flux<CoreEvent>> setTracingContext(ProfilingService profilingService, String artifactId,
+                                                                      String artifactType,
+                                                                      ComponentLocation componentLocation) {
+      if (profilingService != null) {
+        flux =
+            flux.doOnNext(coreEvent -> setTracingContext(profilingService, artifactId, artifactType, componentLocation,
```

Comment:
should this be done as part of the chain of the processing strategy? Isn't other way without modifying the chain?